### ISSUE TITLE
Scrollbar. Crash "scrollOffset should be non-negative"

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-jb#114]
1. open `codeviewer` example
2. change compose version to "0.2.0-build124"
3. run
4. open big file
5. scroll to the middle with scrollbar, then scroll to the beginning

```
Exception in thread "AWT-EventQueue-0 @coroutine#2" java.lang.IllegalArgumentException: scrollOffset should be non-negative (-229723)
	at androidx.compose.foundation.lazy.ItemRelativeScrollPosition.update-uGq0bfM(LazyListState.kt:543)
	at androidx.compose.foundation.lazy.LazyListState$snapToItemIndex$2.invokeSuspend(LazyListState.kt:209)
	at androidx.compose.foundation.lazy.LazyListState$snapToItemIndex$2.invoke(LazyListState.kt)
	at androidx.compose.foundation.gestures.ScrollableController$scroll$2.invokeSuspend(Scrollable.kt:203)
	at androidx.compose.foundation.gestures.ScrollableController$scroll$2.invoke(Scrollable.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:194)
	at androidx.compose.foundation.gestures.ScrollableController.scroll(Scrollable.kt:195)
	at androidx.compose.foundation.lazy.LazyListState.snapToItemIndex(LazyListState.kt:208)
	at androidx.compose.foundation.LazyScrollbarAdapter.scrollTo(Scrollbar.kt:403)
	at androidx.compose.foundation.SliderAdapter$position$1.invokeSuspend(Scrollbar.kt:461)
	(Coroutine boundary)
	at androidx.compose.foundation.SliderAdapter$position$1.invokeSuspend(Scrollbar.kt:461)
	(Coroutine creation stacktrace)
	at kotlin.coroutines.intrinsics.IntrinsicsKt__IntrinsicsJvmKt.createCoroutineUnintercepted(IntrinsicsJvm.kt:122)
	at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:26)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at androidx.compose.foundation.SliderAdapter.setPosition(Scrollbar.kt:460)
	at androidx.compose.foundation.ScrollbarKt$Scrollbar$1$dragObserver$1.onDrag-k-4lQ0M(Scrollbar.kt:218)
	at androidx.compose.ui.gesture.RawDragGestureFilter.onPointerEvent-d1fqKvQ(RawDragGestureFilter.kt:352)
	at androidx.compose.ui.input.pointer.Node.dispatchToPointerInputFilter(HitPathTracker.kt:535)
	at androidx.compose.ui.input.pointer.Node.dispatchChanges(HitPathTracker.kt:435)
	at androidx.compose.ui.input.pointer.Node.dispatchChanges(HitPathTracker.kt:431)
	at androidx.compose.ui.input.pointer.Node.dispatchChanges(HitPathTracker.kt:431)
	at androidx.compose.ui.input.pointer.Node.dispatchChanges(HitPathTracker.kt:431)
	at androidx.compose.ui.input.pointer.Node.dispatchChanges(HitPathTracker.kt:431)
	at androidx.compose.ui.input.pointer.Node.dispatchChanges(HitPathTracker.kt:431)
	at androidx.compose.ui.input.pointer.Node.dispatchChanges(HitPathTracker.kt:431)
	at androidx.compose.ui.input.pointer.Node.dispatchChanges(HitPathTracker.kt:431)
	at androidx.compose.ui.input.pointer.NodeParent.dispatchChanges(HitPathTracker.kt:285)
	at androidx.compose.ui.input.pointer.HitPathTracker.dispatchChanges(HitPathTracker.kt:132)
	at androidx.compose.ui.input.pointer.PointerInputEventProcessor.process-EhkylT0(PointerInputEventProcessor.kt:72)
	at androidx.compose.ui.platform.DesktopOwner.processPointerInput(DesktopOwner.kt:248)
	at androidx.compose.ui.platform.DesktopOwners.onMouseDragged(DesktopOwners.kt:127)
	at androidx.compose.desktop.ComposeWindow$initCanvas$3$mouseDragged$1.invoke(ComposeWindow.kt:122)
	at androidx.compose.desktop.ComposeWindow$initCanvas$3$mouseDragged$1.invoke(ComposeWindow.kt)
	at androidx.compose.desktop.AWTDebounceEventQueue$job$1.invokeSuspend(AWTDebounceEventQueue.kt:58)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:389)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.lang.IllegalArgumentException: scrollOffset should be non-negative (-229723)
	at androidx.compose.foundation.lazy.ItemRelativeScrollPosition.update-uGq0bfM(LazyListState.kt:543)
	at androidx.compose.foundation.lazy.LazyListState$snapToItemIndex$2.invokeSuspend(LazyListState.kt:209)
Caused by: java.lang.IllegalArgumentException: scrollOffset should be non-negative (-229723)
	at androidx.compose.foundation.lazy.LazyListState$snapToItemIndex$2.invoke(LazyListState.kt)
	at androidx.compose.foundation.gestures.ScrollableController$scroll$2.invokeSuspend(Scrollable.kt:203)
	at androidx.compose.foundation.gestures.ScrollableController$scroll$2.invoke(Scrollable.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:194)
	at androidx.compose.foundation.gestures.ScrollableController.scroll(Scrollable.kt:195)
	at androidx.compose.foundation.lazy.LazyListState.snapToItemIndex(LazyListState.kt:208)
	at androidx.compose.foundation.LazyScrollbarAdapter.scrollTo(Scrollbar.kt:403)
	at androidx.compose.foundation.SliderAdapter$position$1.invokeSuspend(Scrollbar.kt:461)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:84)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	... 36 more

```

**Comment by igordmn** [comment]
Fixed in 0.2.0-build129

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5617">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
